### PR TITLE
Fix datatype of attribute metadata

### PR DESCRIPTION
**Pull request by mrolli**
#### Pull Request (PR) description
By adding datatypes in 4.1.0 the code now breaks as the attribute $metadata of class mrepo::repo actually does not only support a single string but also an array of strings. The strings are comprise of a set of keywords allowed to use.

#### This Pull Request (PR) fixes the following issues
n/a

### CODE REVIEW

File: manifests/repo.pp [hunk 1]
```diff
@@ -118,23 +118,23 @@
...
-  Enum['std', 'ncc', 'rhn'] $type           = 'std',
-  Optional[String[1]] $typerelease          = undef,
-  Optional[String[1]] $mrepo_env            = undef,
-  Stdlib::Absolutepath $mrepo_command       = '/usr/bin/mrepo',
-  String $mrepo_options                     = '-qgu',
-  Optional[String[1]] $mrepo_logging        = undef,
-  Optional[String[1]] $ncc_username         = undef,
-  Optional[String[1]] $ncc_password         = undef,
+  Hash[String, String] $urls                                                                            = {},
+  Variant[Enum['yum','apt','repomd','repoview'],Array[Enum['yum','apt','repomd','repoview']]] $metadata = 'repomd',
```
Comment (Dan33l) [inline manifests/repo.pp]: Like with bellow parameter `update`, it will be clearer to create a custom data type.

Comment (genebean) [reply manifests/repo.pp]: @mrolli - are you game to do the adjustment mentioned here? If not I am happy to do it.

Comment (mrolli) [reply manifests/repo.pp]: @genebean Hope this is what you meant. Didn't squash by intention to make the addition of the new datatype explicit.

Comment (alexjfisher) [reply manifests/repo.pp]: I'm pretty sure it's what he had in mind.  Merged!  Thanks for the fix.

**Pull request merged**